indices/algos.py

```diff
@@ -0,0 +1,92 @@
+from networkx import DiGraph
+
+
+def cites_downstream(in_doi: str, graph: DiGraph, out_dois: set) -> bool:
+    """
+    Checks whether `in_doi` cites (has an out_edge pointing to) any papers in `out_dois`.
+
+    Arguments
+    ---------
+    in_doi: The doi to check
+    graph: The graph containing all citations
+    out_dois: The set of dois that may be cited by `in_doi`
+
+    Returns
+    -------
+    is_cited: True if `in_doi` cites something in `out_doi`, otherwise False
+    """
+    citations = 0
+    for _, in_target in graph.out_edges(in_doi):
+        if in_target in out_dois:
+            return True
+    return False
+
+def count_papers_citing(dois: set, graph: DiGraph) -> int:
+    """
+    Get the papers citing the given set of dois.
+
+    Arguments
+    ---------
+    dois: The dois whose citations should be listed
+    graph: The graph containing all citations
+
+    Returns
+    -------
+    citations: The number of articles citing the given dois
+    """
+    papers_citing = set()
+    for doi in dois:
+        for source, _ in graph.in_edges(doi):
+            papers_citing.add(source)
+    return len(papers_citing)
+
+def disruption_index(doi: str, graph: DiGraph) ->  float:
+    """
+    Calculates the disruption index from Wu, Wang, and Evans.
+
+    In short, the disruption index is the difference between the number of the articles citing
+    `doi` but none of its children and the number of articles citing both `doi` and at least
+    one of its children.
```

Comment:
Might be useful to give some intuition for what the results mean - maybe you could add something like "Values are between 1 and -1: a positive disruption index indicates a "disrupting" paper, and a negative disruption index indicates a "developing" paper."

Reply:
Good call, will update the docstring